core/translate/node.rs

```diff
@@ -0,0 +1,428 @@
+use core::fmt;
+use std::{
+    fmt::{Display, Formatter},
+    rc::Rc,
+};
+
+use sqlite3_parser::ast;
+
+use crate::{function::AggFunc, schema::BTreeTable, util::normalize_ident, Result};
+
+/**
+  A query plan node.
+  Nodes form a tree structure, with each node having zero or more children.
+  For example, a query like `SELECT t1.foo FROM t1 ORDER BY t1.foo LIMIT 1` would have the following structure:
+  ```
+    Limit
+      Order
+        Project
+          Scan
+  ```
+*/
+#[derive(Clone, Debug)]
+pub enum QueryPlanNode {
```

Comment:
I think this enum represents logical query plan operators. Therefore, I would perhaps call it `Operator` and maybe call this file `plan.rs`.